taigaio/taiga
# [back] feat(notifications): u#2899 notifications scaffolding + notify when a story is [un]assigned

<div align="center">

![](https://media.giphy.com/media/XEVPg1irhETydAG1tf/giphy.gif)

</div>

Notification scaffolding (first functional version) and 2 endpoints to list the authenticated user's notifications.
```
GET /my/notifications[?read=true|false]
GET /my/notifications/count
```

Added the first two notifications when a story is assigned and when is unassigned.

Some fixes/refactors:
- Some typos in code comments
- Remove `@unwrap_typer_param` because it generates an error with the command parameters (cannot be used) and it's already solved in the latest versions of mypy
- Minor refactor in `__modify_schema__()` for serializers fields.


## How to test:
- [x] Roses are red and test are green.
  - Run the backend and the frontend as user1, user2 and user3
  - [x] Assign
    - As user1 create a story S1
    - As user2 assign S1 to user3
    - What should happen?
      - There are two notifications, one for user1 (creator) and other for user1 (assigned)
      - There are two events, one for every notification created.
  - [x] Unassign
    -  As user2 unassign user3 from S1
    - What should happen?
      - There are two notifications, one for user1 (creator) and other for user1 (unassigned)
      - There are two events, one for every notification created.
